From 16490ed6f8eb761f5b80db6aebed6d805e48f57c Mon Sep 17 00:00:00 2001
From: Heiko Voigt <hvoigt@hvoigt.net>
Date: Wed, 3 Feb 2010 19:46:42 +0100
Subject: [PATCH] spawning of processes: combine for loops and resolve fixme

This enables easier handling of quoting the arguments. Additionally we
can get rid of the FIXME comment in the source.

Signed-off-by: Heiko Voigt <hvoigt@hvoigt.net>
---
 msys/rt/src/winsup/cygwin/spawn.cc |    6 +-----
 1 files changed, 1 insertions(+), 5 deletions(-)

diff --git a/msys/rt/src/winsup/cygwin/spawn.cc b/msys/rt/src/winsup/cygwin/spawn.cc
index e7278e6..108b2f1 100644
--- a/msys/rt/src/winsup/cygwin/spawn.cc
+++ b/msys/rt/src/winsup/cygwin/spawn.cc
@@ -515,11 +515,7 @@ spawn_guts (HANDLE hToken, const char * prog_arg, const char *const *argv,
 	  debug_printf("newargv[%d] = %s", i, newargv[i]);
 	  newargv.replace (i, tmpbuf);
 	  free (tmpbuf);
-	}
-      FIXME; // Is the below loop necessary?
-      // Well at least combine the above with it.
-      for (int i = 0; i < newargv.argc; i++)
-	{
+
 	  char *p = NULL; // Temporary use pointer.
 	  const char *a; // Pointer to newargv element.
 
-- 
1.6.5.1.msysgit.1.12.gf4ccd

